influxql/v1tests/server_test.go

```diff
@@ -4718,12 +4720,75 @@ func TestServer_Query_ShowMeasurements(t *testing.T) {
 		fmt.Sprintf(`other,host=server03,region=caeast value=100 %d`, mustParseTime(time.RFC3339Nano, "2009-11-10T23:00:00Z").UnixNano()),
 	}
 
+
+	ctx := context.Background()
+
+	client := s.MustNewAdminClient()
+	bucket2 := influxdb.Bucket{
+		OrgID:               s.DefaultOrgID,
+		Name: "b2",
+	}
+	bucket3 := influxdb.Bucket{
+		OrgID:               s.DefaultOrgID,
+		Name: "b3",
+	}
+	require.NoError(t, client.CreateBucket(ctx, &bucket2))
+	require.NoError(t, client.CreateBucket(ctx, &bucket3))
+
+	require.NoError(t, client.DBRPMappingService.Create(ctx, &influxdb.DBRPMapping{
+		Database:        "db0",
+		RetentionPolicy: "rp1",
+		Default:         false,
+		OrganizationID:  s.DefaultOrgID,
+		BucketID:        bucket2.ID,
+	}))
+
+	require.NoError(t, client.DBRPMappingService.Create(ctx, &influxdb.DBRPMapping{
+		Database:        "db1",
+		RetentionPolicy: "rp0",
+		Default:         false,
+		OrganizationID:  s.DefaultOrgID,
+		BucketID:        bucket3.ID,
+	}))
+
+
+	rp1Writes := []string{
+		fmt.Sprintf(`other2,host=server03,region=caeast value=100 %d`, mustParseTime(time.RFC3339Nano, "2009-11-10T23:00:00Z").UnixNano()),
+	}
+
+	db1Writes := []string{
+		fmt.Sprintf(`cpu,host=server03,region=caeast value=100 %d`, mustParseTime(time.RFC3339Nano, "2009-11-10T23:00:00Z").UnixNano()),
+		fmt.Sprintf(`disk,host=server03,region=caeast value=100 %d`, mustParseTime(time.RFC3339Nano, "2009-11-10T23:00:00Z").UnixNano()),
+	}
+
 	test := NewTest("db0", "rp0")
 	test.writes = Writes{
 		&Write{data: strings.Join(writes, "\n")},
+		&Write{bucketID: bucket2.ID, data: strings.Join(rp1Writes, "\n")},
+		&Write{bucketID: bucket3.ID, data: strings.Join(db1Writes, "\n")},
 	}
 
 	test.addQueries([]*Query{
+		{
+			name:    `show measurements`,
+			command: "SHOW MEASUREMENTS",
+			// *unlike* 1.x, InfluxDB 2 shows measurements from the default retention policy when the rp is not specified, not all retention policies
```

Comment:
I think this is a bug - https://docs.influxdata.com/influxdb/v2.0/query-data/influxql/ says influxql should be exactly the same as 1.x . I think we should fix this to return measurements from all retention policies mapped for a given db, similar to 1.x .